crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -143,7 +151,7 @@ impl Ethereum {
             access_list: Some(tx.access_list.into()),
             // Specifically set high gas because some nodes don't pick a sensible value if omitted.
             // And since we are only interested in access lists a very high value is fine.
-            gas: Some(MAX_BLOCK_SIZE.into()),
+            gas: Some(self.max_block_size()),
```

Comment:
Conceptual question: If we set gas too high for the simulation, but the max gas the user is willing to pay is less, the simulation will pass but the settlement will fail?

Reply:
If I understand correctly, the gas costs are shared among all participants in the auction. If there's a significant spike in gas prices, it could impact the output amounts, potentially causing some trades to fail if they fall outside the user's specified range. However, this is less likely to happen compared to situations where the gas limit is too low.